src/main/kotlin/ai/digital/integration/server/release/tasks/cluster/operator/OperatorBasedUpgradeReleaseClusterTask.kt

```diff
@@ -0,0 +1,150 @@
+package ai.digital.integration.server.release.tasks.cluster.operator
+
+import ai.digital.integration.server.common.cluster.operator.AwsEksHelper
+import ai.digital.integration.server.common.cluster.operator.AwsOpenshiftHelper
+import ai.digital.integration.server.common.cluster.operator.GcpGkeHelper
+import ai.digital.integration.server.common.cluster.operator.OperatorHelper
+import ai.digital.integration.server.common.constant.K8sSetup
+import ai.digital.integration.server.common.constant.PluginConstant
+import ai.digital.integration.server.common.constant.ProductName
+import ai.digital.integration.server.common.util.GitUtil
+import ai.digital.integration.server.common.util.XlCliUtil
+import ai.digital.integration.server.release.tasks.cluster.ReleaseClusterUtil
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import org.gradle.kotlin.dsl.withGroovyBuilder
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+
+open class OperatorBasedUpgradeReleaseClusterTask : DefaultTask() {
```

Comment:
99% it is a copy paste of `OperatorBasedUpgradeDeployClusterTask`, we can introduce a parent class for it.